DREAM3DReviewFilters/GenerateMaskFromSimpleShapes.cpp

```diff
@@ -0,0 +1,669 @@
+//
+// -----------------------------------------------------------------------------
+void GenerateMaskFromSimpleShapes::createImageMask()
+{
+  size_t totalNumFIDs = m_CentersPtr.lock()->getNumberOfTuples();
+  size_t totalNumElementsDest = m_MaskPtr.lock()->getNumberOfTuples();
+  DataContainer::Pointer dc = getDataContainerArray()->getDataContainer(getMaskArrayPath().getDataContainerName());
+  ImageGeom::Pointer image = dc->getGeometryAs<ImageGeom>();
+  SizeVec3Type udims = image->getDimensions();
+  FloatVec3Type uorigin = image->getOrigin();
+  FloatVec3Type uspacing = image->getSpacing();
+
+  int64_t dims[3] = {
+      static_cast<int64_t>(udims[0]),
+      static_cast<int64_t>(udims[1]),
+      static_cast<int64_t>(udims[2]),
+  };
+
+  float origin[3] = {
+      static_cast<float>(uorigin[0]),
+      static_cast<float>(uorigin[1]),
+      static_cast<float>(uorigin[2]),
+  };
+
+  float spacing[3] = {
+      static_cast<float>(uspacing[0]),
+      static_cast<float>(uspacing[1]),
+      static_cast<float>(uspacing[2]),
+  };
+
+  bool inBounds = false;
+
+  float currentx = 0;
+  float currenty = 0;
+  float currentz = 0;
+
+  int64_t xindex = 0;
+  int64_t yindex = 0;
+  int64_t zindex = 0;
+  int64_t tempMod = 0;
+
+  size_t k = 0;
+
+  if(m_MaskShape == 0)
+  {
+    for(size_t i = 0; i < totalNumElementsDest; i++)
+    {
+
+      zindex = i / (dims[0] * dims[1]);
+      tempMod = i % (dims[0] * dims[1]);
+      yindex = tempMod / (dims[0]);
+      xindex = tempMod % (dims[0]);
+
+      currentx = origin[0] + spacing[0] * xindex;
+      currenty = origin[1] + spacing[1] * yindex;
+      currentz = origin[2] + spacing[2] * zindex;
+
+      inBounds = false;
+      k = 0;
+
+      while(!inBounds && k < totalNumFIDs)
+      {
+        inBounds = IsPointInEllipsoidBounds(m_Centers[3 * k], m_Centers[3 * k + 1], m_Centers[3 * k + 2], m_AxisLengths[3 * k], m_AxisLengths[3 * k + 1], m_AxisLengths[3 * k + 2], currentx, currenty,
+                                            currentz);
+
+        if(inBounds)
+        {
+          m_Mask[i] = true;
+        }
+
+        k++;
+      }
+    }
+  }
+
+  if(m_MaskShape == 1)
+  {
+    for(size_t i = 0; i < totalNumElementsDest; i++)
+    {
+
+      zindex = i / (dims[0] * dims[1]);
+      tempMod = i % (dims[0] * dims[1]);
+      yindex = tempMod / (dims[0]);
+      xindex = tempMod % (dims[0]);
+
+      currentx = origin[0] + spacing[0] * xindex;
+      currenty = origin[1] + spacing[1] * yindex;
+      currentz = origin[2] + spacing[2] * zindex;
+
+      inBounds = false;
+      k = 0;
+
+      while(!inBounds && k < totalNumFIDs)
+      {
+        inBounds = IsPointInCylinderBounds(m_Centers[3 * k], m_Centers[3 * k + 1], m_Centers[3 * k + 2], m_CylinderRad[k], m_CylinderHeight[k], currentx, currenty, currentz);
+
+        if(inBounds)
+        {
+          m_Mask[i] = true;
+        }
+
+        k++;
+      }
+    }
+  }
+
+  if(m_MaskShape == 2)
```

Comment:
Is this case supposed to be different for the previous one?

Reply:
line 313 should be this (which will address this issue): 

        inBounds = IsPointInBoxBounds(m_Centers[3 * k], m_Centers[3 * k + 1], m_Centers[3 * k + 2], m_BoxDims[0], m_BoxDims[1], m_BoxDims[2], currentx, currenty, currentz);

Reply:
pushed in latest commit